### PR TITLE
Extract RPC summaries from backend responses.

### PR DESCRIPTION
**Pull request by jkohen**
Implemented CreateTimeSeriesSummary parser. A follow-up change will use this to produce metrics.

### CODE REVIEW

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
You may have to rebase your branch on the latest `stackdriver-agent-5.8.1` — it's still not quite stable, and underwent another rebase. It should be simple in an interactive rebase: just delete https://github.com/Stackdriver/collectd/pull/158/commits/6a0aef8032d0a191ea35bdcaec3cc5618105e76d and https://github.com/Stackdriver/collectd/pull/158/commits/8a2a4f8fab29d9cfce42d330b8ea58f6209234b8, and replay the rest of the commits.

**Comment by jkohen** [comment]
> You may have to rebase your branch on the latest `stackdriver-agent-5.8.1` — it's still not quite stable, and underwent another rebase. It should be simple in an interactive rebase: just delete [6a0aef8](https://github.com/Stackdriver/collectd/commit/6a0aef8032d0a191ea35bdcaec3cc5618105e76d) and [8a2a4f8](https://github.com/Stackdriver/collectd/commit/8a2a4f8fab29d9cfce42d330b8ea58f6209234b8), and replay the rest of the commits.

Thanks for the warning. Can you provide the commands I need to run to do the rebase? Any idea when it'll become stable so that a simple `git merge` works?

**Comment by igorpeshansky** [comment]
I was suggesting an [interactive rebase](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase):
```
git fetch && git checkout stackdriver-agent-5.8.1 && git reset --hard origin/stackdriver-agent-5.8.1
git checkout rpc_response
git rebase -i stackdriver-agent-5.8.1
# In the rebase-todo, drop the first two commits with numbers 6a0aef8 and 8a2a4f8 and save.
# You should have a clean rebase at this point. Check it with:
git log rpc_response..stackdriver-agent-5.8.1  # should be empty
```
@qingling128 and I just agreed to freeze the current `stackdriver-agent-5.8.1` branch history, so this was the last rebase.

**Comment by bmoyles0117** [comment]
I'll wait for the rebase to review, apparently the DIFF shows changes that are not part of your changes. When you've done the rebase, make sure you do `git push -f` to force the push.

**Comment by jkohen** [comment]
> I was suggesting an [interactive rebase](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase):
> 
> ```
> git fetch && git checkout stackdriver-agent-5.8.1 && git reset --hard origin/stackdriver-agent-5.8.1
> git checkout rpc_response
> git rebase -i stackdriver-agent-5.8.1
> # In the rebase-todo, drop the first two commits with numbers 6a0aef8 and 8a2a4f8 and save.
> # You should have a clean rebase at this point. Check it with:
> git log rpc_response..stackdriver-agent-5.8.1  # should be empty
> ```
> 
> @qingling128 and I just agreed to freeze the current `stackdriver-agent-5.8.1` branch history, so this was the last rebase.

Done, thank you. I didn't see 8a2a4f8 though.

**Comment by qingling128** [comment]
This one seems ready for review. @bmoyles0117 @igorpeshansky

**Comment by bmoyles0117** [comment]
Reviewing now, thanks for the ping @qingling128

File: src/collectd_time_series_response_test.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+      "index": 0,
+      "error": {
+        "code": 429,
+        "status": "RESOURCE_EXHAUSTED",
+        "message": "Retry in 300 milliseconds."
+      },
+      "value_errors": [
+        {
+        "index": 0
```
Comment (bmoyles0117) [inline src/collectd_time_series_response_test.json]: nit: indentation seems off, can you run these files through JSON lint?

File: src/utils_stackdriver_json.c [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    field_t current_field;
+    // Temporary storage for building error instances.
+    time_series_error_t temp_error;
+  } state;
+
+  // Holds the output.
+  time_series_summary_t *response;
+} parse_summary_t;
+
+static void print_context(const parse_summary_t *ctx) {
```
Comment (bmoyles0117) [inline src/utils_stackdriver_json.c]: Can we rename this to debug_context or log_context? I initially thought this was being printed unconditionally.

**Review by bmoyles0117 — APPROVED** [review]
LGTM, barring 2 small comments.

File: src/collectd_time_series_response_test.json (hunk 1 above)
Comment (jkohen) [reply src/collectd_time_series_response_test.json]: Done, with `json_pp`

**Comment by jkohen** [comment]
Submitting. Thanks, Bryan!

**Pull request merged**